### PR TITLE
Changed all occurences of NULL to nullptr

### PR DESCRIPTION
**Pull request by BodilessSleeper**
Fixes issue #147

### CODE REVIEW

**Comment by stiber** [comment]
We have just gone through major merges of large changes in the code base, so this will have to be reviewed very carefully. Files have moved, files have been renamed, classes have been merged/split/created, and class and variable names have been changed.

File: Simulator/Core/BGDriver.cpp [hunk 1]
```diff
@@ -137,7 +137,7 @@ int main(int argc, char *argv[]) {
 
    // INPUT OBJECTS ARENT IN PROJECT YET
    // Terminate the stimulus input
-//   if (pInput != NULL)
+//   if (pInput != nullptr)
```
Comment (stiber) [inline Simulator/Core/BGDriver.cpp:140]: File name has changed.

Comment (BodilessSleeper) [reply Simulator/Core/BGDriver.cpp:140]: Eh, you could've merged this PR first and then #170 and we would've had no problems, but I guess I'll just redo the problematic files (not sure if this is the only one of all the changed files)

**Comment by BodilessSleeper** [comment]
Should be done now

**Review by stiber — APPROVED** [review]

**Pull request merged**